src/nvim/api/autocmd.c

```diff
@@ -280,9 +282,29 @@ Array nvim_get_autocmds(Dict(get_autocmds) *opts, Error *err)
           PUT(autocmd_info, "desc", CSTR_TO_OBJ(ac->desc));
         }
 
-        PUT(autocmd_info,
-            "command",
-            STRING_OBJ(cstr_as_string(aucmd_exec_to_string(ac, ac->exec))));
+        switch (ac->exec.type) {
+        case CALLABLE_CB: {
+          Callback *cb = &ac->exec.callable.cb;
+          switch (cb->type) {
+          case kCallbackLua:
+            if (nlua_ref_is_function(cb->data.luaref)) {
+              PUT(autocmd_info, "callback", LUAREF_OBJ(api_new_luaref(cb->data.luaref)));
+            }
+            break;
+          default:
```

Comment:
```suggestion
          case kCallbackFuncref:
          case kCallbackPartial:
```

and add a `case kCallbackNone:` that just `asserts(false);` or something to signal that it should never happen.